access_request_handler.go

```diff
@@ -89,6 +91,30 @@ func (f *Fosite) NewAccessRequest(ctx context.Context, r *http.Request, session
 	}
 	accessRequest.Client = client
 
+	// PKCE Handling
```

Comment:
I really think this should be a custom handler and not hardcoded here

Reply:
Most of the code are (re)generated mocks ^^

Reply:
About PKCE settings (RequiredForPublicClient), maybe via a decorator (sort of middleware) in the authorize and token endpoints.

This could be helpful to add / process parameters before handling core OAuth operations.

Reply:
The idea of this framework is to have handlers [such as this one for the auth endpoint](https://github.com/ory/fosite/blob/master/handler/oauth2/flow_authorize_code_auth.go) which implement parts of the oauth2 spec. What you would do (probably) is some checks at the `/auth` endpoint and then some checks or updates to the response at the [`/token` endpoint](https://github.com/ory/fosite/blob/master/handler/oauth2/flow_authorize_code_token.go)